protocol-designer/src/components/organisms/StepSummary/AbsorbanceReaderSummary.tsx

```diff
@@ -0,0 +1,122 @@
+import { useTranslation } from 'react-i18next'
+
+import {
+  ALIGN_CENTER,
+  Flex,
+  SPACING,
+  StyledText,
+  Tag,
+} from '@opentrons/components'
+
+import {
+  ABSORBANCE_READER_INITIALIZE,
+  ABSORBANCE_READER_LID,
+  ABSORBANCE_READER_READ,
+} from '../../../constants'
+import { StyledTrans } from './StyledTrans'
+
+import type { RobotState } from '@opentrons/step-generation'
+import type { FormData } from '../../../form-types'
+
+interface AbsorbanceReaderSummaryProps {
+  currentStep: FormData
+  labwareNicknamesById: Record<string, string>
+  labwareState: RobotState['labware']
+}
+
+export function AbsorbanceReaderSummary(
+  props: AbsorbanceReaderSummaryProps
+): JSX.Element | null {
+  const { t } = useTranslation('protocol_steps')
+  const { currentStep, labwareNicknamesById, labwareState } = props
+  const {
+    moduleId: absorbanceReaderModuleId,
+    absorbanceReaderFormType,
+    fileName,
+    lidOpen,
+    mode,
+    referenceWavelength,
+    referenceWavelengthActive,
+    wavelengths,
+  } = currentStep
+  const labwareOnAbsorbanceReaderId = Object.values(
+    labwareState
+  ).find(({ stack }) => stack.includes(absorbanceReaderModuleId as string))
+    ?.stack[0] // top element of stack including absorbance reader
+  const labwareOnAbsorbanceReaderNickname =
+    labwareOnAbsorbanceReaderId != null
+      ? labwareNicknamesById[labwareOnAbsorbanceReaderId]
+      : null
+  let stepSummaryContent: JSX.Element | null = null
+  switch (absorbanceReaderFormType) {
+    case ABSORBANCE_READER_READ: {
+      stepSummaryContent = (
+        <StyledTrans
+          i18nKey="protocol_steps:absorbance_reader.read"
+          tagText={labwareOnAbsorbanceReaderNickname ?? undefined}
+          tagText2={`${fileName}`}
+        />
+      )
+      break
+    }
+    case ABSORBANCE_READER_INITIALIZE: {
+      const wavelengthsComponents = (
+        <Flex gridGap={SPACING.spacing4}>
+          {wavelengths.map((wavelength: number, index: number) => (
+            <>
```

Comment:
probably wavelength wouldn't be good? 
anyway need to add `key`.

```suggestion
            <React.Fragment key={`wavelength-${index}`}>
```

Reply:
thanks, good call!